### PR TITLE
SaveDuringPlay - Test type string equality instead of contains

### PR DESCRIPTION
**Pull request by gaborkb**
[Delete any line or section that does not apply]

### Purpose of this PR

Made SaveDuringPlay string with equality instead of contains.

### Testing status

[Explanation of what’s tested, how tested and existing or new automation tests. Can include manual testing by self and/or QA. Specify test plans. Rarely acceptable to have no testing.]

- [ ] Added an automated test
- [ ] Passed all automated tests
- [x] Manually tested 

### Documentation status

[Overview of how documentation is affected by this change. If there is no effect on documentation, explain why. Otherwise, state which sections are changed and why.]

- [ ] Updated [CHANGELOG](https://keepachangelog.com/en/1.0.0/)
- [ ] Updated README (if applicable)
- [ ] Commented all public classes, properties, and methods
- [ ] Updated user documentation

### Samples status

- [ ] Re-packed Extras~/CinemachineExamples.unitypackage

### Technical risk

[Overall product level assessment of risk of change. Need technical risk & halo effect.]

### Comments to reviewers

[Info per person for what to focus on, or historical info to understand who have previously reviewed and coverage. Help them get context.]

### Package version

[Justification for updating either the patch, minor, or major version according to the [semantic versioning](https://semver.org/spec/v2.0.0.html) rules]

- [ ] Updated package version

### CODE REVIEW

**Review by glabute — APPROVED** [review]

**Comment by gaborkb** [comment]
Approving your changes. I manually tested yours, and then also manually tested my changes.

File: Editor/Utility/SaveDuringPlay.cs [hunk 1]
```diff
@@ -557,7 +551,6 @@ static void RestoreAllInterestingStates()
                     Undo.RegisterFullObjectHierarchyUndo(go, "SaveDuringPlay");
                     if (saver.PutFieldValues(go, roots))
                     {
-                        //Debug.Log("SaveDuringPlay: updated settings of " + saver.ObjetFullPath);
```
Comment (glabute) [inline Editor/Utility/SaveDuringPlay.cs:558]: actually, I'd leave this line in (but commented out).  Very useful for debugging, and I've had occasion to uncomment it several times

File: Editor/Utility/SaveDuringPlay.cs [hunk 2]
```diff
@@ -385,28 +383,19 @@ static object LeafObjectFromString(Type type, string value, GameObject[] roots)
...
+                    return c == null ? string.Empty : ObjectTreeUtil.GetFullName(c.gameObject);
+                case GameObject go:
+                    return ObjectTreeUtil.GetFullName(go);
+                case ScriptableObject so:
+                    return AssetDatabase.GetAssetPath(so);
+                default:
+                    return obj.ToString();
             }
-            return obj.ToString();
         }
```
Comment (glabute) [inline Editor/Utility/SaveDuringPlay.cs]: I'm uneasy about this change.  I'm wondering if it properly handles Unity's overriding of the == null operator (which will return true for non-null objects that have been deleted).  I'm doubtful that the c# hocus-pocus here will handle those cases the same way.

**Review by glabute — CHANGES_REQUESTED** [review]

File: Editor/Utility/SaveDuringPlay.cs (hunk 2 above)
Comment (gaborkb) [reply Editor/Utility/SaveDuringPlay.cs]: Ok. I checked and it looked like it handles "null" cases too, but to be sure I reverted it back to the original state.

**Review by glabute — APPROVED** [review]

**Pull request merged**